generator/src/main/java/com/scottlogic/deg/generator/generation/VelocityMonitor.java

```diff
@@ -7,28 +7,34 @@
 import com.scottlogic.deg.generator.walker.reductive.FixedField;
 import com.scottlogic.deg.generator.walker.reductive.ReductiveState;
 
+import java.math.BigDecimal;
 import java.math.BigInteger;
-import java.text.SimpleDateFormat;
-import java.util.Date;
+import java.math.RoundingMode;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.Map;
 import java.util.Timer;
 import java.util.TimerTask;
 
 public class VelocityMonitor implements ReductiveDataGeneratorMonitor {
-    private String startedGenerating;
+    private static final BigDecimal millisecondsInSecond = BigDecimal.valueOf(1_000);
+    private static final BigDecimal nanoSecondsInMillisecond = BigDecimal.valueOf(1_000_000);
+
+    private LocalDateTime startedGenerating;
     private long rowsSinceLastSample;
     private BigInteger rowsEmitted;
     private Timer timer = new Timer(true);
-    private SimpleDateFormat simpleDateFormat = new SimpleDateFormat("HH:mm:ss");
+    private DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HH:mm:ss");
     private long previousVelocity = 0;
 
     @Override
     public void generationStarting(GenerationConfig generationConfig) {
-        startedGenerating = simpleDateFormat.format( new Date());
+        startedGenerating = LocalDateTime.now();
```

Comment:
Is it safer to use UTC time instead of local? i.e. you are generating over a time when the clocks change or your locale erroneously changes between start timing and end timing

Reply:
Its not a huge issue i think, the user might expect to see times in local format, but would support duration analysis more accurately; changed to `OffsetDateTime`